src/sidebar/components/HypothesisApp.tsx

```diff
@@ -62,6 +65,22 @@ function HypothesisApp({
     }
   }, [isSidebar, profile, settings, store]);
 
+  const shareDialogTabs = (() => {
+    const tabs: ShareDialogTab[] = [];
+
+    if (!isThirdPartyService(settings)) {
+      tabs.push('share');
+    }
+    if (store.isFeatureEnabled('export_annotations')) {
+      tabs.push('export');
+    }
+    if (store.isFeatureEnabled('import_annotations')) {
+      tabs.push('import');
+    }
+
+    return tabs;
+  })();
```

Comment:
I initially wrapped this in `useMemo`, but `isFeatureEnabled` returns `false` until the features are asynchronosly loaded, and for some reason, making `store` a dependency of `useMemo` was not making it run again, so export and import tabs were never displayed.

This runs the logic in every render, which is not ideal. I'm trying to figure out why the above didn't work.

Reply:
> making store a dependency of useMemo was not making it run again, so export and import tabs were never displayed.

The `store` value never changes. What you need to do instead is extract the store value outside the `useMemo` call and then make it a dependency, like so:

```js
const fooEnabled = store.isFeatureEnabled('bar');
const derivedValue = useMemo(() => { /* do something with `fooEnabled` */} , [fooEnabled]);
```

This is an unfortunate hazard of the way that the store API interacts with `useMemo`.

Reply:
Right. Well, that's actually reasonable.